server/libs/amqp.js

```diff
@@ -0,0 +1,95 @@
+const amqp = require('amqplib/callback_api');
+const BSON = require('bson');
+const db = require('./database');
+const wss = require('./websocket').websocket
+const bson = new BSON();
+
+/**
+ * Get column names and store in global variable.
+ */
+console.log('Updating column names...');
+let columnNames = '';
+db.pool.connect(function(err, client, done) {
+  if (err) {
+    return console.error(db.errors.CONNECT_ERROR, err.stack);
+  }
+  const query = 'SELECT column_name FROM information_schema.columns ' +
+                'WHERE table_schema = \'public\' AND table_name = \'packet\'';
+  client.query(query, function(err, result) {
+    done();
+    if (err) {
+      return console.error(db.errors.SELECT_ERROR, err.stack);
+    }
+
+    // build column name list and remove trailing comma and whitespace
+    for (let i = 1; i < result.rows.length; i++) {
+      columnNames += result.rows[i].column_name + ', ';
+    }
+    columnNames = columnNames.replace(/,\s*$/, '');
+  });
+});
+console.log('Column names updated');
+
+/**
+ * Setup the AMQP channel with RabbitMQ (change host if needed)
+ */
+const rmq_URL = 'amqp://localhost:5672';
+amqp.connect(rmq_URL, function(err, conn) {
+  try {
+    conn.createChannel(function(err, ch) {
```

Comment:
Are promises supported? That would look nicer compared to having a huge try catch block.